compiler/rustc_ast/src/util/comments.rs

```diff
@@ -42,10 +43,27 @@ pub fn beautify_doc_string(data: Symbol) -> Symbol {
         if i != 0 || j != lines.len() { Some((i, j)) } else { None }
     }
 
-    fn get_horizontal_trim(lines: &[&str]) -> Option<usize> {
+    fn get_horizontal_trim(lines: &[&str], kind: CommentKind) -> Option<usize> {
         let mut i = usize::MAX;
         let mut first = true;
 
+        // In case we have doc comments like `/**` or `/*!`, we want to remove stars if they are
```

Comment:
This isn't really removing stars, though. It's removing blank lines.

The for loop below this conditional removes stars, as well as spaces and tabs.

What you really do here is removing blank lines, so that the code below will actually remove the stars.

Reply:
I badly worded it. Good catch!

Reply:
I extended the comment a bit.